File: math/concatenate-non-zero-digits-and-multiply-by-sum-i.cpp
class Solution {
public:
    long long sumAndMultiply(int n) {
        long long x = 0;
        long long sum = 0;

        // Extract digits in order
        string s = to_string(n);

        for(char c : s){
            if(c != '0'){
                int d = c - '0';
                x = x * 10 + d; // build number
                sum += d;      // sum of digits
            }
        }

        // If no non-zero digits, x remains 0 and sum is 0 → return 0
        return x * sum;
    }
};

File: 4135-concatenate-non-zero-digits-and-multiply-by-sum-i/concatenate-non-zero-digits-and-multiply-by-sum-i.java
class Solution {
    public long sumAndMultiply(int n) {
        long ans=0;
        long sum=0;
        int m=n;
        int rev=0;
        while(m>0){
            int d=m%10;
            if(d!=0){
                sum+=d;
                rev=rev*10+d;
            }
            m/=10;
        }
        int k=0;
        while(rev>0){
            int d=rev%10;
            k=k*10+d;
            rev/=10;
        }
        ans=k*sum;
        return ans;
    }
}